rfcs/0004-fulltext-search.md

```diff
@@ -0,0 +1,223 @@
+# RFC-0000: Template
+
+<dl>
+  <dt>Author</dt>
+  <dd>Ford Nickels</dd>
+
+  <dt>RFC pull request</dt>
+  <dd><a href="https://github.com/graphprotocol/rfcs/pull/11">URL</a></dd>
+
+  <dt>Obsoletes (if applicable)</dt>
+  <dd>-</dd>
+
+  <dt>Date of submission</dt>
+  <dd>2020-01-05</dd>
+
+  <dt>Date of approval</dt>
+  <dd>-</dd>
+
+  <dt>Approved by</dt>
+  <dd>-</dd>
+</dl>
+
+## Contents
+
+<!-- toc -->
+
+## Summary
+
+This RFC proposes adding a fulltext search filter type to the GraphQL API 
+to allow subgraph developers to specify language-specific, lexical, 
+composite filters that end users can use in their queries. The fulltext 
+index will be created by examining all words in a document, breaking it 
+into individul words and phrases (lexical analysis), and collapsing 
+variations of words into a single index term (stemming.)
+
+## Goals & Motivation
+
+The current set of string filters available in our GraphQL API is lacking 
+the tools needed to build effective, modern interfaces. Wildcard string 
+matching does provide string filtering, but users have come to expect the 
+easy to use filtering that comes with fulltext search systems.
+
+To facilitate building effective user interfaces we must provide human-user 
+friendly query filtering. Lexical, composite fulltext search filters can 
+provide the tools necessary for front-end developers to implement powerful 
+search bars that filter data across multiple fields of an Entity.
+
+With the proposed feature I aim to:
+  - provide tools for subgraph developers to define composite search 
```

Comment:
I think fulltext search indexes should be able to span across entities so this language is a little limiting.

Reply:
I think full text search could be very useful across entities, so I've updated this language to include that intent. 
With that goal in mind I'm working on how to enable this behavior in the GraphQL API while still encouraging data models that will give performant querying and index building.